pkg/controller/workload/job/job_controller.go

```diff
@@ -207,10 +219,25 @@ func (r *JobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.R
 		}
 		log.V(3).Info("Job is suspended and workload not yet admitted by a clusterQueue, nothing to do")
 		return ctrl.Result{}, nil
+	} else {
+		log.Info("Job is running", "enabled", r.waitForPodsReady, "ready", jobPodsReady(&job))
+		// handle an unsuspended job when waitForPodsReady is enabled
+		if r.waitForPodsReady && jobPodsReady(&job) {
```

Comment:
Maybe this logic should be in line 249

Reply:
Also, we should add the condition with Status=false if we are still waiting.

Reply:
Good idea. It allows to also distinguish two situations: (1) the job controller has not inspected the Job, (2) the job controller already inspected the job but the pods are not ready.

Is it possible that the workload is in PodsReady condition already, but a user increases the level of parallelism. If so, should we keep the condition or recompute it? Currently I recompute.